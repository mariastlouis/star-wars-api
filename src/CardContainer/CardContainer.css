@import url("https://fonts.googleapis.com/css?family=Pathway+Gothic+One");
@import url("https://fonts.googleapis.com/css?family=Cabin");
body {
  background: black;
  color: white;
  font-family: "Cabin", sans-serif; }

h1 {
  font-size: 20px; }

h2 {
  display: block;
  -webkit-margin-before: 0;
  -webkit-margin-after: 0;
  -webkit-margin-start: 0px;
  -webkit-margin-end: 0px; }

.card-container {
  display: flex;
  flex-direction: row;
  margin-left: auto;
  margin-right: auto;
  width: 1000px; }

.card-holder {
  display: flex;
  flex-wrap: wrap; }

.category {
  margin-bottom: 20px;
  text-align: center; }

@media screen and (max-width: 908px) {
  .card-container {
    width: 750px; } }

@media screen and (max-width: 750px) {
  .card-container {
    width: 500px; } }

@media screen and (max-width: 520px) {
  .card-container {
    width: 250px; } }
