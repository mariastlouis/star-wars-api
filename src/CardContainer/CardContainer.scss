@import 'variable';

.card-container {
  display: flex;
  flex-direction: row;
  margin-left: auto;
  margin-right: auto;
  width: 1000px;
}

.card-holder {
  display: flex;
  flex-wrap: wrap;
}

.category {
  margin-bottom: 20px;
  text-align: center;
}


@media screen and (max-width: 908px) {
  .card-container {
    width: 750px;
  }
}

@media screen and (max-width: 750px) {
  .card-container { 
    width: 500px;
  }
}

@media screen and (max-width: 520px) {
  .card-container { 
    width: 250px;
  }
}