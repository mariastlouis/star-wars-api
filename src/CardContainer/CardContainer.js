import React from 'react';
import Card from '../Card/Card.js'
import PropTypes from 'prop-types';
import './CardContainer.scss';



const CardContainer = ({characterArray}) => {



  return (
    <div className = "cardContainer">
      {
        characterArray.map((character, index) =>{
          return (

            <Card
            name = {character.name}
            data = {character.data} />
            )
        })
      }
      
    </div>
  )
}

export default CardContainer;