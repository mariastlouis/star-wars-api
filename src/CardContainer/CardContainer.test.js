import React from 'react';
import CardContainer from './CardContainer.js';
import {shallow} from 'enzyme';

const mockAddFavorite = jest.fn();

const mockVehicleInfo = [
  {
    name: 'Sand Crawler',
    favorite: false,
    info: {
      Class: 'Wheeled',
      Model: 'Digger Crawler',
      Passengers: '30'
    }
  },
  {
    name: "X-34 landspeeder",
    favorite: true,
    info: {
      Class: "repulsorcraft",
      Model: "X-34 landspeeder",
      Passengers: "1"
    }
  }
];

const mockCategory = 'vehicle';

let wrapper;

describe('Card Container', () => {
  beforeEach(() => {
   
    wrapper = shallow(
      <CardContainer 
        category = {mockCategory}
        info = {mockVehicleInfo}
        addFavorite = {mockAddFavorite}
      />);
  });

  it('should render correctly', () =>{
    expect(wrapper).toBeDefined();
  });

  it('should match the snapshot', () =>{
    expect(wrapper).toMatchSnapshot();
  });

  it('should render 2cards if the array length is 2', () =>{
    expect(wrapper.find('Card').length).toEqual(2);
  });

});