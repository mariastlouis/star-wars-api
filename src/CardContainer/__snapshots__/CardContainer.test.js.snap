// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Container should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardContainer
    addFavorite={[Function]}
    category="vehicle"
    info={
        Array [
            Object {
              "favorite": false,
              "info": Object {
                "Class": "Wheeled",
                "Model": "Digger Crawler",
                "Passengers": "30",
              },
              "name": "Sand Crawler",
            },
            Object {
              "favorite": true,
              "info": Object {
                "Class": "repulsorcraft",
                "Model": "X-34 landspeeder",
                "Passengers": "1",
              },
              "name": "X-34 landspeeder",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="card-holder"
>
        <Card
                addFavorite={[Function]}
                category="vehicle"
                favorite={false}
                info={
                        Object {
                                "Class": "Wheeled",
                                "Model": "Digger Crawler",
                                "Passengers": "30",
                              }
                }
                name="Sand Crawler"
        />
        <Card
                addFavorite={[Function]}
                category="vehicle"
                favorite={true}
                info={
                        Object {
                                "Class": "repulsorcraft",
                                "Model": "X-34 landspeeder",
                                "Passengers": "1",
                              }
                }
                name="X-34 landspeeder"
        />
</div>,
      "className": "card-container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            addFavorite={[Function]}
            category="vehicle"
            favorite={false}
            info={
                        Object {
                                    "Class": "Wheeled",
                                    "Model": "Digger Crawler",
                                    "Passengers": "30",
                                  }
            }
            name="Sand Crawler"
/>,
          <Card
            addFavorite={[Function]}
            category="vehicle"
            favorite={true}
            info={
                        Object {
                                    "Class": "repulsorcraft",
                                    "Model": "X-34 landspeeder",
                                    "Passengers": "1",
                                  }
            }
            name="X-34 landspeeder"
/>,
        ],
        "className": "card-holder",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "addFavorite": [Function],
            "category": "vehicle",
            "favorite": false,
            "info": Object {
              "Class": "Wheeled",
              "Model": "Digger Crawler",
              "Passengers": "30",
            },
            "name": "Sand Crawler",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "addFavorite": [Function],
            "category": "vehicle",
            "favorite": true,
            "info": Object {
              "Class": "repulsorcraft",
              "Model": "X-34 landspeeder",
              "Passengers": "1",
            },
            "name": "X-34 landspeeder",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="card-holder"
>
          <Card
                    addFavorite={[Function]}
                    category="vehicle"
                    favorite={false}
                    info={
                              Object {
                                        "Class": "Wheeled",
                                        "Model": "Digger Crawler",
                                        "Passengers": "30",
                                      }
                    }
                    name="Sand Crawler"
          />
          <Card
                    addFavorite={[Function]}
                    category="vehicle"
                    favorite={true}
                    info={
                              Object {
                                        "Class": "repulsorcraft",
                                        "Model": "X-34 landspeeder",
                                        "Passengers": "1",
                                      }
                    }
                    name="X-34 landspeeder"
          />
</div>,
        "className": "card-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Card
              addFavorite={[Function]}
              category="vehicle"
              favorite={false}
              info={
                            Object {
                                          "Class": "Wheeled",
                                          "Model": "Digger Crawler",
                                          "Passengers": "30",
                                        }
              }
              name="Sand Crawler"
/>,
            <Card
              addFavorite={[Function]}
              category="vehicle"
              favorite={true}
              info={
                            Object {
                                          "Class": "repulsorcraft",
                                          "Model": "X-34 landspeeder",
                                          "Passengers": "1",
                                        }
              }
              name="X-34 landspeeder"
/>,
          ],
          "className": "card-holder",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "function",
            "props": Object {
              "addFavorite": [Function],
              "category": "vehicle",
              "favorite": false,
              "info": Object {
                "Class": "Wheeled",
                "Model": "Digger Crawler",
                "Passengers": "30",
              },
              "name": "Sand Crawler",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "function",
            "props": Object {
              "addFavorite": [Function],
              "category": "vehicle",
              "favorite": true,
              "info": Object {
                "Class": "repulsorcraft",
                "Model": "X-34 landspeeder",
                "Passengers": "1",
              },
              "name": "X-34 landspeeder",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
